### ISSUE TITLE
redundant whitespace nodes

### ISSUE THREAD

**Issue by borkdude** [carocad/parcera#1]
It seems to me that the `[:whitespace ""]` nodes aren't necessary:

```
user=> (parcera/clojure "{:a 1}")
[:code [:whitespace ""] [:map [:map-content [:whitespace ""] [:simple-keyword "a"] [:whitespace " "] [:whitespace ""] [:number "1"] [:whitespace ""]]] [:whitespace ""]]
```

Why is the result wrapped in a `[:code ...]` vector? Can it be something else than code?

**Comment by carocad** [comment]
> It seems to me that the [:whitespace ""] nodes aren't necessary:
@borkdude yeah, as we discussed in slack they are not necessary but otherwise Instaparse would be too slow. I have tried several alternatives to deal with whitespace but so far none of them had better performance ... unless we find a way to make the optional whitespace parsing faster, it will stay like that.

> Why is the result wrapped in a [:code ...] vector? Can it be something else than code?
Instaparse requires a "root node" to start the parsing process; in this case that is ":code". I can of course hide it but that would mean that you will end up with something like `[[:list ...][:list ....]]` which is not consistent with the rest as one vector has `[:keyword & content]` while another one has `[ & content]`. I thought about it and went with consistency over "prettyness" 😅
